File: src/test/java/no/nav/pensjon/selvbetjeningopptjening/security/masking/MaskerTest.java
package no.nav.pensjon.selvbetjeningopptjening.security.masking;

import no.nav.pensjon.selvbetjeningopptjening.opptjening.Pid;
import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.*;

class MaskerTest {

    @Test
    void maskFnr_masks_other_fnr_digits_than_birthDate() {
        assertEquals("123456*****", Masker.maskFnr("12345678901"));
        assertEquals("049253*****", Masker.maskFnr(new Pid("04925398980")));
    }

    @Test
    void maskFnr_masks_all_digits_in_nonFnr() {
        assertEquals("****** (length 10)", Masker.maskFnr("1234567890"));
    }

    @Test
    void maskFnr_handles_empty_string() {
        assertEquals("****** (length 0)", Masker.maskFnr(""));
    }

    @Test
    void maskFnr_returns_textNull_for_valueNull() {
        assertEquals("null", Masker.maskFnr((Pid) null));
        assertEquals("null", Masker.maskFnr((String) null));
    }
}

File: src/test/java/no/nav/pensjon/selvbetjeningopptjening/tech/security/masking/MaskerTest.kt
package no.nav.pensjon.selvbetjeningopptjening.tech.security.masking

import io.kotest.core.spec.style.FunSpec
import io.kotest.matchers.shouldBe
import no.nav.pensjon.selvbetjeningopptjening.opptjening.Pid

class MaskerTest : FunSpec({

    test("maskFnr masks other fnr digits than birthDate") {
        Masker.maskFnr("12345678901") shouldBe "123456*****"
        Masker.maskFnr(Pid("04925398980")) shouldBe "049253*****"
    }

    test("maskFnr masks all digits in nonFnr") {
        Masker.maskFnr("1234567890") shouldBe "****** (length 10)"
    }

    test("maskFnr removes control characters") {
        Masker.maskFnr("\n0492539\r8980\t") shouldBe "049253*****"
    }

    test("maskFnr handles empty string") {
        Masker.maskFnr("") shouldBe "****** (length 0)"
    }

    test("maskFnr returns textNull for valueNull") {
        Masker.maskFnr(null as Pid?) shouldBe "null"
        Masker.maskFnr(null as String?) shouldBe "null"
    }
})
